### PR TITLE
MicroFocus Contributed - Improve efficiency by reducing calls to AccuRev server

### PR DESCRIPTION
**Pull request by PoojaVishal**
While Polling from Accurev, too many hist calls in Accurev polling logs.
I compressed the hist command with all transaction types in single command with date range to get all transactions happened after last build and checked for changes found.

Also i have made changes to History parser to parse all transaction properly which initially used to give null and duplicate values of transaction.

### CODE REVIEW

**Comment by casz** [comment]
https://github.com/jenkinsci/accurev-plugin/commit/e35294c3c37618191642088550e6dd215f0a32d5.patch
https://github.com/jenkinsci/accurev-plugin/commit/c8ab863f3878d77dae43adf89d34d142d1933231.patch

@PoojaVishal 
You might want to set up an email alias on your GitHub account 🥇 
Or look into fixing your git user.email config 😅 
to get proper credit for your commits 🏆 
https://help.github.com/articles/adding-an-email-address-to-your-github-account/
https://help.github.com/articles/setting-your-commit-email-address-in-git
then to amend your commits: https://stackoverflow.com/a/3042512

File: src/main/resources/jenkins/plugins/accurev/AccurevTool/config.groovy [hunk 1]
```diff
@@ -11,4 +11,4 @@ f.entry(field: "name", title: _("Name")) {
 }
 f.entry(field: "home", title: _("Path to AccuRev executable")) {
   f.textbox()
-}
+}
```
Comment (casz) [inline src/main/resources/jenkins/plugins/accurev/AccurevTool/config.groovy]:
```suggestion
}

```

File: src/main/java/hudson/plugins/accurev/parsers/xml/ParseHistory.java [hunk 2]
```diff
@@ -44,27 +44,33 @@ public Boolean parse(XmlPullParser parser, List<AccurevTransaction> context)
...
+        resultTransaction.setDate(convertAccurevTimestamp(parser.getAttributeValue("", "time")));
+        resultTransaction.setUser(parser.getAttributeValue("", "user"));
+        eventType = parser.nextTag();
+
+        // loop the parent tag elements until we reach the end of the parent tag
+        while (eventType == XmlPullParser.START_TAG) {
+          // check the children tags
+          if ("comment".equals(parser.getName())) {
+            resultTransaction.setMsg(parser.nextText());
+            System.out.println("poo" + resultTransaction.getMsg());
```
Comment (casz) [inline src/main/java/hudson/plugins/accurev/parsers/xml/ParseHistory.java]: Random println
```suggestion
```

Comment (PoojaVishal) [reply src/main/java/hudson/plugins/accurev/parsers/xml/ParseHistory.java]: My bad!
I have pushed the change

**Comment by casz** [comment]
Please consider rewording the pull request title as it is used for release notes

**Comment by PoojaVishal** [comment]
Hi Joseph,

There are few more changes i did in code. Please review and let me know.
I also reworded the commit message.

**Comment by PoojaVishal** [comment]
i also configured commit email address.

**Comment by PoojaVishal** [comment]
@casz Kindly see if this PR title is ok.

**Comment by casz** [comment]
Suggestion as `AccuRev Plugin` is redundant in this context.
`improve efficiency by reducing calls to AccuRev server`

**Comment by casz** [comment]
Is the new hist command only supported on newer version of AccuRev server?

**Comment by PoojaVishal** [comment]
@casz  No this will work for older accurev server also.

**Comment by PoojaVishal** [comment]
@casz I was trying to squash all commits.
the pull request is closed.
Did i do something wrong?

**Comment by PoojaVishal** [comment]
Thank god..:)

**Comment by casz** [comment]
You can always see the commit history here: https://github.com/jenkinsci/accurev-plugin/commits/master

**Comment by PoojaVishal** [comment]
@casz  we have a scheduled release on oct 14th.
Once we are done with final round of testing, i will remind you to do a public release on 14th.

**Comment by PoojaVishal** [comment]
@casz The build is still running. Its saying node windows 8 offline.
Does it usually takes this long?

**Comment by PoojaVishal** [comment]
Hi Joseph,

The xml pull parser is throwing EOFException in parse history class in Linux but works fine in windows..
Do u have any idea if this parser is pathform dependent?
We are unable to release because of this issue..☹️

**Comment by PoojaVishal** [comment]
@casz  you were right, this combined hist command is supported from AccuRev 7.X version not older.

**Pull request merged**